java/src/main/java/org/rocksdb/StringAppendOperator.java

```diff
@@ -18,6 +20,19 @@ public StringAppendOperator(char delim) {
         super(newSharedStringAppendOperator(delim));
     }
 
+    public StringAppendOperator(byte[] delim) {
+        super(newSharedStringAppendTESTOperator(delim));
+    }
+
+    public StringAppendOperator(String delim) {
```

Comment:
I think we should avoid the constructor without the `charset`.

Reply:
I do not mind deleting it. I was just following usual shortcut idea of Java standard packages where this kind of methods use `Charset.defaultCharset()` if `charset` is not passed. Do you think that the user awareness about `charset` should be explicitly forced?